tests/unit/swagger/templateHelpers.spec.ts

```diff
@@ -172,34 +172,60 @@ describe('ValidationService', () => {
     });
 
     describe('Enum validate', () => {
+        const name = 'name';
+        type Enumeration = string[] | number[];
+        interface FieldErrors { [key: string]: any; }
+        let validateEnum: (value: any, enumeration: Enumeration, fieldErrors?: FieldErrors) => any;
+
+        beforeEach(() => {
+          const validationService = new ValidationService({});
+          validateEnum = (
+            value: any,
+            enumeration: Enumeration,
+            fieldErrors?: FieldErrors,
+          ) => validationService.validateEnum(
+            name, value, fieldErrors || {}, enumeration);
+        });
+
         it('should enum number value', () => {
-            const value = 1;
-            const result = new ValidationService({}).validateEnum('name', value, {}, ['0', '1'] as any);
+            const value = '1';
+            const result = validateEnum(value, ['0', '1']);
             expect(result).to.equal(value);
         });
 
         it('should enum string value', () => {
             const value = 'HELLO';
-            const result = new ValidationService({}).validateEnum('name', value, {}, ['HELLO'] as any);
+            const result = validateEnum(value, ['HELLO']);
             expect(result).to.equal(value);
         });
 
         it('should enum no member', () => {
             const error: any = {};
-            const name = 'name';
             const value = 'HI';
-            const result = new ValidationService({}).validateEnum(name, value, error, [] as any);
+            const result = validateEnum(value, [], error);
             expect(result).to.equal(undefined);
             expect(error[name].message).to.equal(`no member`);
         });
 
         it('should enum out of member', () => {
             const error: any = {};
-            const name = 'name';
             const value = 'SAY';
-            const result = new ValidationService({}).validateEnum(name, value, error, ['HELLO', 'HI'] as any);
+            const result = validateEnum(value, ['HELLO', 'HI'], error);
             expect(result).to.equal(undefined);
-            expect(error[name].message).to.equal(`should be one of the following; ['HELLO', 'HI']`);
+            expect(error[name].message).to.equal(
+              `should be one of the following; ['HELLO', 'HI']`);
+        });
+
```

Comment:
@drdgvhbh We need a unit test that verifies that the following would be an error:

Given the following enum, the validation service should return an error if the value provided was a non-numeric string like “foo”.

```ts
export enum MyIntEnum {
  one = 1
  two = 2
}
```